apps/helpers/lib/hasher.ex

```diff
@@ -38,11 +38,13 @@ defmodule Hasher do
     end
   end
 
+  @spec update_hash(binary() | :error, binary() | :error) :: binary() | :error
   defp update_hash(_, :error), do: :error
   defp update_hash(:error, _), do: :error
   defp update_hash(chunk, hash) when is_binary(chunk), do: :crypto.hash_update(hash, chunk)
```

Comment:
why is the `when is_binary(chunk)` needed?

Reply:
I don’t think it is needed. Neither should the first parameter ever be :error, but I’m a bit afraid to touch it. I don’t trust that much the typing system. What do you think?

Reply:
Meh. Let’s be optimistic about the stream library and dialyzer ;)